### PR TITLE
Add more options to DescribeE

### PR DESCRIPTION
**Pull request by zhangguanzhang**
fix https://github.com/go-rod/rod/issues/89

Description for the changes.
ref
https://chromedevtools.github.io/devtools-protocol/1-3/DOM/#method-describeNode

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
Golang's int64 zero is 0, but the code I submitted didn't change the Depth, so I don't know why the unit test failed

**Comment by ysmood** [comment]
please add at least one test.

**Comment by zhangguanzhang** [comment]
等我有时间了加上单元测试，到时候再合并吧

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,12 @@
 /lib/assets/node_modules
 tmp/
 coverage.txt
+
```
Comment (Hamzantal) [inline .gitignore:4]: these files should be part of your global .gitignore. generally ide/system items shouldn't be defined for each project

Comment (zhangguanzhang) [reply .gitignore:4]: Goland is one of the hot GO development tools, and these ignore lists are necessary

Comment (Hamzantal) [reply .gitignore:4]: Do you know what a global .gitignore is? 
**IDE and System items should not be defined seperately for each project**.

Comment (zhangguanzhang) [reply .gitignore:4]: If you open a project using Goland, the `.ide`a folder will be generated under the root of the project

File: fixtures/describe.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<html>
+<body>
+<ul>
+  <li>coffee</li>
+  <li>tee
```
Comment (Hamzantal) [inline fixtures/describe.html:5]: the correct spelling is `tea`

File: .gitignore (hunk 1 above)
Comment (Hamzantal) [reply .gitignore:4]: I know what the .idea folder is. I use Goland as well. But that shouldn't stop this project from following these guidelines.

**Comment by ysmood** [comment]
As you mentioned, now you can use `Elements` to resolve #89, I feel we can close this PR.

If you can add two extra parameters `depth` and `pierce` to the `DescribeE`, we can reconsider it, for example: `page.Element("div").DescribeE(-1, true)`

@Hamzantal `describeDepth` is a bad design to just pass a parameter to a function 😂 , we should never allow this kind of design to enter our codebase.

**Comment by ysmood** [comment]
If you have any questions, please feel free to talk with us.

**Comment by zhangguanzhang** [comment]
`page.Element("div").DescribeE(-1, true)` 
This breaks previous compatibility and requires rewriting many of the calls

**Comment by ysmood** [comment]
Sure, but we are under `v1.0.0`, from the https://semver.org/ definition, we can break API. I already talked about our policy of API stability in the readme: https://github.com/go-rod/rod#q-how-versioning-is-handled

**Comment by zhangguanzhang** [comment]
Well, the second value  which `true` represents what behavior

**Comment by ysmood** [comment]
Use your ide, go to the definition, you will see it `proto.DOMDescribeNode`, use your ide more

**Comment by zhangguanzhang** [comment]
I will try later

**Comment by ysmood** [comment]
Thank you!

**Comment by zhangguanzhang** [comment]
and `Describe(depth int, pierce bool)` also need to?

**Comment by ysmood** [comment]
I don't think so, you can read [E suffix section](https://github.com/go-rod/rod/blob/master/.github/CONTRIBUTING.md#e-suffix) to understand why.

I feel seldom people will need to use it, most time they will use `Elements`

**Comment by zhangguanzhang** [comment]
ok

**Comment by ysmood** [comment]
You won't have to rewrite a lot of code, you should find out it's really easy to contribute code for rod

File: element.go [hunk 3]
```diff
@@ -223,7 +234,7 @@ func (el *Element) DescribeE() (*proto.DOMNode, error) {
 
 // ShadowRootE returns the shadow root of this element
 func (el *Element) ShadowRootE() (*Element, error) {
-	node, err := el.DescribeE()
+	node, err := el.DescribeE(-1, true)
```
Comment (ysmood) [inline element.go:237]: To get the shadow element, we don't need the whole tree.

File: element.go [hunk 4]
```diff
@@ -241,7 +252,7 @@ func (el *Element) ShadowRootE() (*Element, error) {
 
 // FrameE doc is similar to the method Frame
 func (el *Element) FrameE() (*Page, error) {
-	node, err := el.DescribeE()
+	node, err := el.DescribeE(-1, true)
```
Comment (ysmood) [inline element.go:255]: To get the frame, we don't need the whole tree.

File: sugar.go [hunk 5]
```diff
@@ -422,7 +422,7 @@ func (k *Keyboard) InsertText(text string) {
 // Describe returns the element info
 // Returned json: https://chromedevtools.github.io/devtools-protocol/tot/DOM#type-Node
 func (el *Element) Describe() *proto.DOMNode {
-	node, err := el.DescribeE()
+	node, err := el.DescribeE(-1, true)
```
Comment (ysmood) [inline sugar.go:425]: By default, I feel it's better to use `(1, false)`, most time we just use it to get the current node info, not the tree. Or can you give a strong reason why we should return huge data that may impact performance by default?

File: element.go (hunk 3 above)
Comment (zhangguanzhang) [reply element.go:237]: use this  `(1, false)`

Comment (ysmood) [reply element.go:237]: yes

**Comment by zhangguanzhang** [comment]
The github CI error occurred again, and I passed without modifying the trigger 😜

**Pull request merged**